### PR TITLE
PHP Warning:  count()

### PR DESCRIPTION
**Pull request by copperhead57**
Clean up Warning messages
PHP Warning:  count(): Parameter must be an array or an object that implements Countable in ......\videodb\core\functions.php on line 456.

As of php 7.3 the above is being signaled as a warning. add code to use is_countable function to check variable.
For pre php 7.3 users added code to emulate the is_countable function (can be removed once videodb is moved to minimum php version 7.3

Note: as of php 8 the warning will become an error level.

### CODE REVIEW

File: core/functions.php [hunk 1]
```diff
@@ -452,9 +452,19 @@ function getActorThumbnail($name, $actorid = 0, $idSearchAllowed = true)
 	// identify actor by unique actor id, of by name
     if ($actorid && $idSearchAllowed) {
         $result = runSQL($SQL." WHERE actorid='".escapeSQL($actorid)."'");
-	}
-	if (!$actorid || (count($result) == 0)) {
```
Comment (johanneskonst) [inline core/functions.php:456]: I would keep this bit and just add an is_array($result) check. No need for the is_countable bit...
is_countable is good for modern code but the runSQL function will only return an array (when all succeeds), so just check on that:
`if (!$actorid || (is_array($result) && count($result) == 0)) {`

**Review by johanneskonst — CHANGES_REQUESTED** [review]
As commented inline...

File: core/functions.php (hunk 1 above)
Comment (copperhead57) [reply core/functions.php:456]: agree change in progress

Comment (copperhead57) [reply core/functions.php:456]: @johanneskonst I have made one small change to your suggestion, have group the is_array and count() in brackets for clarity.
if you agree I will do PR with squash and merge to clean up number of commits to master

**Pull request merged**